diff -rupN vlv_kernel_old/drivers/gpu/drm/Makefile vlv_kernel/drivers/gpu/drm/Makefile
--- vlv_kernel_old/drivers/gpu/drm/Makefile	2012-10-29 14:04:48.557054214 +0800
+++ vlv_kernel/drivers/gpu/drm/Makefile	2012-10-29 14:17:26.409999437 +0800
@@ -34,6 +34,7 @@ obj-$(CONFIG_DRM_RADEON)+= radeon/
 obj-$(CONFIG_DRM_MGA)	+= mga/
 obj-$(CONFIG_DRM_I810)	+= i810/
 obj-$(CONFIG_DRM_I915)  += i915/
+obj-$(CONFIG_DRM_EMGD)  += emgd/
 obj-$(CONFIG_DRM_MGAG200) += mgag200/
 obj-$(CONFIG_DRM_CIRRUS_QEMU) += cirrus/
 obj-$(CONFIG_DRM_SIS)   += sis/
